crates/tree/compute_bin_stats.rs

```diff
@@ -163,89 +163,37 @@ pub fn compute_bin_stats_subtraction(
 	}
 }
 
-unsafe fn compute_bin_stats_column_major_root_no_hessians<T>(
-	gradients: &[f32],
-	binned_feature_values: &[T],
-	bin_stats_for_feature: &mut [BinStatsEntry],
-) where
-	T: ToPrimitive,
-{
-	let len = gradients.len();
-	for i in 0..len {
-		let ordered_gradient = *gradients.get_unchecked(i);
-		let bin_index = binned_feature_values.get_unchecked(i).to_usize().unwrap();
-		let bin_stats = bin_stats_for_feature.get_unchecked_mut(bin_index);
-		bin_stats.sum_gradients += ordered_gradient as f64;
-		bin_stats.sum_hessians += 1.0;
-	}
-}
-
-pub unsafe fn compute_bin_stats_column_major_root_yes_hessians<T>(
+unsafe fn compute_bin_stats_column_major_root<T, const HESSIANS: bool>(
 	gradients: &[f32],
 	hessians: &[f32],
 	binned_feature_values: &[T],
 	bin_stats_for_feature: &mut [BinStatsEntry],
 ) where
 	T: ToPrimitive,
 {
-	let len = gradients.len();
-	for i in 0..len {
-		let ordered_gradient = *gradients.get_unchecked(i);
-		let ordered_hessian = *hessians.get_unchecked(i);
-		let bin_index = binned_feature_values.get_unchecked(i).to_usize().unwrap();
-		let bin_stats = bin_stats_for_feature.get_unchecked_mut(bin_index);
-		bin_stats.sum_gradients += ordered_gradient as f64;
-		bin_stats.sum_hessians += ordered_hessian as f64;
-	}
-}
-
-unsafe fn compute_bin_stats_column_major_not_root_no_hessians<T>(
-	ordered_gradients: &[f32],
-	binned_feature_values: &[T],
-	bin_stats_for_feature: &mut [BinStatsEntry],
-	examples_index: &[u32],
-) where
-	T: ToPrimitive,
-{
-	let len = examples_index.len();
-	#[cfg(target_arch = "x86_64")]
-	let prefetch_len = len.saturating_sub(PREFETCH_OFFSET);
-	#[cfg(not(target_arch = "x86_64"))]
-	let prefetch_len = 0;
-	for i in 0..prefetch_len {
-		#[cfg(target_arch = "x86_64")]
-		{
-			let prefetch_index = *examples_index.get_unchecked(i + PREFETCH_OFFSET);
-			let prefetch_index = prefetch_index.to_usize().unwrap();
-			let prefetch_ptr = binned_feature_values.as_ptr().add(prefetch_index) as *const i8;
-			core::arch::x86_64::_mm_prefetch(prefetch_ptr, core::arch::x86_64::_MM_HINT_T0);
+	if HESSIANS {
+		let len = gradients.len();
+		for i in 0..len {
+			let ordered_gradient = *gradients.get_unchecked(i);
+			let ordered_hessian = *hessians.get_unchecked(i);
+			let bin_index = binned_feature_values.get_unchecked(i).to_usize().unwrap();
+			let bin_stats = bin_stats_for_feature.get_unchecked_mut(bin_index);
+			bin_stats.sum_gradients += ordered_gradient as f64;
+			bin_stats.sum_hessians += ordered_hessian as f64;
+		}
+	} else {
+		let len = gradients.len();
+		for i in 0..len {
+			let ordered_gradient = *gradients.get_unchecked(i);
+			let bin_index = binned_feature_values.get_unchecked(i).to_usize().unwrap();
+			let bin_stats = bin_stats_for_feature.get_unchecked_mut(bin_index);
+			bin_stats.sum_gradients += ordered_gradient as f64;
+			bin_stats.sum_hessians += 1.0;
```

Comment:
instead of duplicating the whole block twice, can just add this after. 
```
if HESSIANS {
  let ordered_hessian = *hessians.get_unchecked(i);
  bin_stats.sum_hessians += ordered_hessian as f64;
} else {
 bin_stats.sum_hessians += 1.0;
}
```

Reply:
Thank you, addressed in [1afc9ee](https://github.com/tangramdotdev/tangram/pull/96/commits/1afc9ee73c3eecf30716266717034ee6e22e6927).